homeassistant/components/plugwise/climate.py

```diff
@@ -192,21 +191,22 @@ def hvac_action(self) -> HVACAction:
         self._previous_action_mode(self.coordinator)
 
         # Adam provides the hvac_action for each thermostat
-        if (control_state := self.device.get("control_state")) == "cooling":
-            return HVACAction.COOLING
-        if control_state == "heating":
-            return HVACAction.HEATING
-        if control_state == "preheating":
-            return HVACAction.PREHEATING
-        if control_state == "off":
-            return HVACAction.IDLE
-
-        heater: str = self.coordinator.data.gateway["heater_id"]
-        heater_data = self.coordinator.data.devices[heater]
-        if heater_data["binary_sensors"]["heating_state"]:
-            return HVACAction.HEATING
-        if heater_data["binary_sensors"].get("cooling_state", False):
-            return HVACAction.COOLING
+        if self._gateway["smile_name"] == "Adam":
+            if (control_state := self.device.get("control_state")) in (
+                HVACAction.COOLING,
+                HVACAction.HEATING,
+                HVACAction.PREHEATING,
+            ):
+                return cast(HVACAction, control_state)
```

Comment:
This seems incorrect. Why would this need a cast?

Upstream isn't using our HVACAction enum right?

Reply:
`control_state` is a `str` coming from the backend, the `cast()` is needed to change it to a HVACAction enum with the single if-statement-with-walrus.

Reply:
That is not correct, it should not be casted. You are now effectively signalling this is an instance of the HVACAction, but it is not. It is a string.

It should return an HVACAction instance instead.

Reply:
Do you prefer the old method with the 3 separate `if`'s? Or should I use match-case?

Reply:
No? Just make an instance of the HVACAction enum using the string?

return HVACAction(mystring)

Reply:
OK, I made a change. Will this do?

Reply:
OK understood :) Will change again.

Reply:
Back to almost original, HVACAction() can't work without a small change in the backend.